File: ZooProjectJava/src/com/zoo/AirBreather.java
package com.zoo;

/**
 * AirBreather is an AirWaterAnimal that breathes air.
 * @author lucaspoon
 *
 */
public class AirBreather extends AirWaterAnimal{
	
	// Used to name airbreather animal.
	static int count = 0;
	public final static String AIR_BREATHER = "AirBreather";
	
	/**
	 *  Default constructor.
	 */
	public AirBreather(){
		this(AIR_BREATHER+count);
		count++;
	}

	/**
	 * Constructor with name.
	 * @param name of animal.
	 */
	public AirBreather(String name) {
		this(name, false, false, false);
	}
	
	/**
	 * Constructor of AirBreather with provided parameters.
	 * @param name of animal.
	 * @param isThirsty value of thirsty.
	 * @param isHungry value of hungry.
	 * @param isAsleep value of asleep.
	 */
	public AirBreather(String name, boolean isThirsty, boolean isHungry, boolean isAsleep){
		super(name, isThirsty, isHungry, isAsleep);
	}
	
	/**
	 * Gets type of animal.
	 * @return animal type.
	 */
	@Override
	protected String getAnimalType() {
		return AIR_BREATHER;
	}

}

File: ZooProjectCpp/src/AirBreather.cpp
/*
 * AirBreather.cpp
 *
 *  Created on: Mar 31, 2019
 *      Author: lucaspoon
 */

#include "AirBreather.h"
#include <iostream>
using namespace std;

int AirBreather::count = 0;
const string AirBreather::AIR_BREATHER = "AirBreather";

AirBreather::AirBreather() : AirWaterAnimal(generateName(),false,false,false){}

AirBreather::AirBreather(string name) : AirWaterAnimal(name, false, false, false){}

AirBreather::AirBreather(string name, bool isThirsty, bool isHungry, bool isAsleep) : AirWaterAnimal(name, isThirsty, isHungry, isAsleep){}

string AirBreather::generateName(){
	string name = AIR_BREATHER;
	name.append(to_string(count));
	count++;
	return name;
}

string AirBreather::getAnimalType() {
	return AIR_BREATHER;
}

AirBreather::~AirBreather() {}

